https://bugs.gentoo.org/544078
https://bugzilla.mindrot.org/show_bug.cgi?id=2369

do not try to generate rsa1 keys when ssh1 support is disabled

--- a/ssh-keygen.c
+++ b/ssh-keygen.c
@@ -948,7 +948,9 @@ do_gen_all_hostkeys(struct passwd *pw)
 		char *key_type_display;
 		char *path;
 	} key_types[] = {
+#ifdef WITH_SSH1
 		{ "rsa1", "RSA1", _PATH_HOST_KEY_FILE },
+#endif
 		{ "rsa", "RSA" ,_PATH_HOST_RSA_KEY_FILE },
 		{ "dsa", "DSA", _PATH_HOST_DSA_KEY_FILE },
 #ifdef OPENSSL_HAS_ECC
